microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/HelidonContainerImpl.java

```diff
@@ -399,10 +399,15 @@ public boolean isRunning() {
 
         @Override
         public BeanManager getBeanManager() {
-            if (isRunning.get()) {
-                return new BeanManagerProxy(beanManager());
+            if (!isRunning.get()) {
+                LOGGER.warning("BeanManager requested during container shutdown. This may be caused by observer methods "
+                                       + "that use CDI.current(). Switch to finest logging to see stack trace.");
+                LOGGER.log(Level.FINEST,
+                           "Invocation of container method during shutdown",
+                           new IllegalStateException("Container not running"));
```

Comment:
Probably not worth it but can be performing better when avoiding `Throwable.fillInStackTrace()`
```java
    if (LOGGER.isLoggable(Level.FINEST)) {
        LOGGER.log(Level.FINEST, "Invocation of container method during shutdown",
                new IllegalStateException("Container not running"));
    }
```

Reply:
Fixed